### PR TITLE
Alloy synthesis

### PR DESCRIPTION
**Pull request by halworsen**
Work in progress.

Adds the alloy synthesizer. Lets you combine metal or glass with a mineral (e.g. diamond, phoron) to make alloys. Different alloys have different benefits, and can be used to create walls, windows and maybe some other stuff with said benefits.
- [x] Alloy synthesizer machine
- [x] Alloy sheets
- [x] Alloy walls/windows/floors
- [x] Benefits from alloys
- [x] Metal + phoron gives plasteel? OR Plasteel is replaced entirely by phorometallic alloy

**Important notes**

This replaces both reinforced walls and plasteel completely! Plasteel is still called plasteel in the code, but now has the name "catametallic alloy".

The new types for them are `/turf/simulated/wall/alloy/reinforced` for reinforced walls and `/obj/item/stack/sheet/alloy/plasteel`. An important distinction is that only catametallic alloys with >= 40% platinum are considered plasteel. Any less than that is considered a normal metallic alloy (`/obj/item/stack/sheet/alloy/metal`).

### CODE REVIEW

**Comment by Kwask** [comment]
Very cool idea! Would love to see this getting expanded with chemistry! :+1:

File: code/game/objects/items/stacks/sheets/alloy.dm [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	singular_name = "alloy sheet"
+	icon = 'icons/obj/items/materials.dmi'
+	icon_state = "polysteel-1"
+	item_state = "polysteel"
+	var/list/materials = list()
+
+/obj/item/stack/sheet/alloy/New(var/list/comp)
+	..()
+	if(!comp)
+		usr << "\red This should never appear. There HAS to be a composition list!"
```
Comment (Kwask) [inline code/game/objects/items/stacks/sheets/alloy.dm]: Make sure not to use text macros like \red for colors, instead use spans like this

``` CSS
<span class='warning'>RED TEXT GOES HERE</span>
```

**Comment by halworsen** [comment]
Need some feedback on the last point. Should plasteel be replaced (if wanted phorometallic alloys could be named plasteel anyways), or should phoron + metal just give plasteel?

**Comment by Kwask** [comment]
Plasteel is platinum-steel, so that should remain as-is, but phoron-steel should exist as well, maybe Phorosteel or something.

Plasteel used to be plasma-steel, but since phoron is now plasma, that name just wouldn't make any sense :confused:

**Comment by nullbear** [comment]
I hate the way phorosteel sounds. Unimaginative me just called it 'supersteel' and stuff.

**Comment by halworsen** [comment]
Plasteel's description claims it's an alloy of steel and phoron, though. Not to mention you can make it by combining phoron and metal in the ore smelting thing at ACE.

**Comment by Kwask** [comment]
If that is true, it's not supposed to be like that. Plasteel is the old name of plasma steel, but ever since plasma was changed to phoron, it shouldn't be that case

**Comment by halworsen** [comment]
Alrighty. But it still leaves the question of whether or not plasteel should be replaced with catametallic alloys (platinum + metal), or the other way around.

**Comment by Kwask** [comment]
Sure, I think it'd help people understand it's a new system if things have new names

**Comment by halworsen** [comment]
How about making it so that different minerals adds to different stats? Apparently goon has alloys, too, and I'm not gonna lie, the idea of handling benefits/disadvantages like this is taken from them. I really like how they handle it, though.

Instead of hardcoding the effects of all the individual minerals, they could add to different stats. This is already done for some minerals, such as phoron, which increases the max temp. The current uranium benefit would be replaced by a rot/radioactivity multiplier or something, and then uranium would increase or decrease those values.

**Comment by Kwask** [comment]
Yeah, ideally that's how it work. Additionally, each different material having different phase states would be cool, so you could use solid chemicals from chemistry in walls and such.

**Comment by halworsen** [comment]
Need to resolve conflicts, add an alloy synthesis lab to the map, and it's ready to merge.

**Comment by Kwask** [comment]
:DDDDDDDDD

Am very excited for this!

**Pull request merged**